### PR TITLE
Autodetect lack of screen presence

### PR DESCRIPTION
**Pull request by TerryMathews**
This is the simplest, most efficient way I could come up with to silence
the "Failed to start write 60" error that occurs when QMK tries to talk
to a screen that doesn't exist.

iota_gfx_init passes a success boolean. We catch that into a global bool
(we could rewrite multiple functions to pass this as an argument, but
given the number of keyboards using this code it seemed less disruptive
this way) and then use that as a conditional on running the
iota_gfx_task in matrix_scan_user.

Tl;dr: if the screen doesn't init, the screen write code doesn't run.

### CODE REVIEW

**Comment by drashna** [comment]
Correct me if I'm wrong, but `iota_gfx_task()` doesn't actually return anything?

**Comment by drashna** [comment]
Ah, okay, I was looking at the wrong iota function then. Don't mind me.

**Comment by TerryMathews** [comment]
@drashna Do you know if there's something wrong with this PR? I had expected it to be merged by now considering how minor it is.

**Comment by jackhumbert** [comment]
I can help ;) thanks!

**Comment by TerryMathews** [comment]
Ty!
